internal/blockprocessor/index_entries.go

```diff
@@ -0,0 +1,284 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package blockprocessor
+
```

Comment:
I think all the index code - the stuff that implements the particular indexing algorithm we plan to use - should be in another package. 
I would also like to see some documentation on the interface, possibly taken from your presentation, to accompany the methods and tests below. These two files should probably be in separate PR, dedicated to index entries, separate from the index db creation, above. Right now it is really difficult to review without context.

Reply:
Agree. I have made this PR to introduce only index creation logic. I have pushed a separate PR for the index DB creation. I am not sure about the interfaces. The index package will provide a function constructIndexEntries() for updates. We are using only the composite key-based index.